### PR TITLE
Refactor MTE-2900 Upgrade Xcode 15.4 for Firefox iOS and Focus iOS

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2900)
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2901)

## :bulb: Description
Let's use the latest stable version of Xcode for the Firefox iOS and Focus iOS work.

Since the source code compiles as is, I can upgrade MacStadium after this change is merged.

Should we do the following?
* Run smoke tests in parallel. Here's potential to save a few minutes: https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/ccb12fcd-74ac-43fd-8381-960c886059c1 (2 simulators running in parallel)

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by clarmso** [comment]
Here are some pipelines with various numbers of parallel simulators:
* No parallel simulators: (21 min): https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/f164ae94-76e0-405f-98ab-02bd8c6bd05a
* 2 parallel simulators (19min): https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/014b4da4-8361-4889-9e21-105c785e7450
* 3 parallel simulator (time out): https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/2ec363b0-9566-4fc8-9462-32c5173c28e2 <-- don't do this

I would not do parallel simulators for now. The overhead of launching parallel simulators is great.

**Comment by clarmso** [comment]
L10nBuild:
https://app.bitrise.io/build/ff1989db-dd00-4abd-bc91-e8677751178e?tab=log

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite2SnapshotTests.swift [hunk 1]
```diff
@@ -108,7 +108,7 @@ class L10nSuite2SnapshotTests: L10nBaseSnapshotTests {
 
     func tapKeyboardKey(_ key: Int) {
         let key = app.keyboards.keys.element(boundBy: key)
-        if app.buttons["Continue"].exists == true {
+        if app.buttons["Continue"].isHittable {
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite2SnapshotTests.swift:111]: I have had many issues with this .isHittable.., let's monitor that tests work (I see you added a link to a build working, but let's keep an eye on this)

**Review by isabelrios — APPROVED** [review]

**Comment by isabelrios** [comment]
> L10nBuild: https://app.bitrise.io/build/ff1989db-dd00-4abd-bc91-e8677751178e?tab=log

If you go to the Generate Screenshots step, there are two tests that failed, the Fakespot one could be flaky, but not sure about the other. Please verify that it is not due to the change added here. Thanks!

**Comment by clarmso** [comment]
> > L10nBuild: https://app.bitrise.io/build/ff1989db-dd00-4abd-bc91-e8677751178e?tab=log
> 
> If you go to the Generate Screenshots step, there are two tests that failed, the Fakespot one could be flaky, but not sure about the other. Please verify that it is not due to the change added here. Thanks!

@isabelrios 😞 The failures are unrelated to the upgrade. Both `main` and this branch's pipeline fails on `test3ReloadButtonContextMenu` and `testFakespotAvailable`. I can't reproduce the failures locally.

Xcode 15.4 (this branch): https://app.bitrise.io/build/b030ef4c-ca1f-48f5-a165-e88f0a027c7e
main: https://app.bitrise.io/build/411481fb-408f-4c49-97de-a833a5fe696d

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite2SnapshotTests.swift (hunk 1 above)
Comment (clarmso) [reply firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite2SnapshotTests.swift:111]: The "Continue" button seems to exist but is covered, so `tap()` within the if-statement does not work at least on my end.

**Pull request merged**